projects/npm-tools/packages/npm-scripts/src/webpack/createFederationConfig.js

```diff
@@ -32,9 +33,14 @@ module.exports = async function () {
 	const name = packageJson.name;
 	const webContextPath = bnd['Web-ContextPath'] || name;
 
-	const {filePath: nullJsFilePath} = createTempFile('null.js', '');
-	const {filePath: mainFilePath} = createTempFile('index.federation.js', '');
+	const {filePath: libInitJsFilePath} = createTempFile('libInit.js', '', {
+		autoDelete: false,
+	});
+	const {filePath: mainFilePath} = createTempFile('index.federation.js', '', {
+		autoDelete: false,
+	});
```

Comment:
To the point about debuggability and leaving temp files around (until the OS reclaims them), is there anywhere in the build output where the actual location is visibly logged? Asking because it matters on macOS where the `tmpdir()` is going to be random unguessable gibberish; the only way to know where to look is if the console shows where the files went (more so if you take my suggestion up above of combining with `mkdtempSync()`).

Reply:
I was thinking in refactoring the temp files stuff a bit, to make it clearer, because since we added the feature to move files apart of source tree (for the `deployFast` command) we changed the paradigm, but the code doesn't reflect it fully (IMO).

So, first, a bit of history...

The `createTempFile` method used to put files in the source tree and delete them on finish to avoid leaving a modified source tree. Because watch tools (specifically `deployFast` + `gradle`) got fooled by that, I created this [config entry](https://github.com/liferay/liferay-frontend-projects/blob/8954af4210985cc17eb0da7a2bd30026809585cc/projects/npm-tools/packages/npm-scripts/src/config/npmscripts.config.js#L176) to move things out of source tree.

That means that:

1. Even though the method is called `createTempFile` it is not really used to create temp files. Or it is, but I like to think of them more as "derived" files (intermediate build artifacts) than the typical random files you would put in `/tmp`. 
2. By default they are written to the build directory (the logical place for intermediate artifacts, probably)
3. There's no need to delete them since they will be removed whenever you `clean` the output.

What can happen now is that someone sees `createTempFile` and thinks that it is used to create true temporary files in the `/tmp` (or equivalent) dir. I don't think we do that in any place in `npm-scripts` now, but if we need it in the future I'm sure we will mess it up :sweat_smile: .

So, my proposal is to rename this method to something else, stop it from autodeleting files and, probably, stop it being configurable and simply put the files in the build directory in a specific place, since they are well known files and, in general, it's very handy to have them available, given the amount of stuff we generate under the hood.

What do you think?

Reply:
In the case of `liferay-portal` what you see after running `gradle deploy` is this:

![imagen](https://user-images.githubusercontent.com/3273630/110598270-d937a900-8181-11eb-9945-bacafcaeb1a1.png)

And specifically the `webpack.config.json` file is full of generated magic options :scream:

Reply:
Yeah, my comment about logging was specifically thinking about debugging _test failures_, where the path may not be so predictable.

> So, my proposal is to rename this method to something else, stop it from autodeleting files and, probably, stop it being configurable and simply put the files in the build directory in a specific place, since they are well known files and, in general, it's very handy to have them available, given the amount of stuff we generate under the hood.
>
> What do you think?

I think _I_ wouldn't change it, but only because I don't remember all the places it is used and wouldn't want to have to analyze them and be sure that it is ok to make such a change. But if you feel confident in doing it and see the value, I wouldn't try to talk you out of it.

Reply:
I think it is almost easy. I tried to set autoDelete to false yesterday and had some problems with the tests, but the code seemed to work correctly for liferay-portal. However it's low priority, so I will try to find a more relaxed moment :-D.